### DISCUSSION TITLE
How to Disable File Download in menu?

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [coder/code-server#4992] [Q&A] [answered]
In VS Code file menu have download option which i need to disable. Can you please guide me to achieve it.

**Comment by lg-nineoneninethreenine** [comment]
I expect it is in the settings, else it is a built-in feature that can't be disabled :thinking: if we could disable it though.

**Reply by iamdineshkumar** [reply]
@im-coder-lg Actually I was not able to find in any settings. This will let others to download files which are sensitive.

**Reply by lg-nineoneninethreenine** [reply]
I think there is a method to disable it, but I don't really know. The download button is used in cloud instances to download a local file. I don't really know how to disable it but @jsjoeio might. If there is no button to disable it, I wonder if we could add one to disable it.

**Reply by iamdineshkumar** [reply]
@jsjoeio  @im-coder-lg Yes its possible just by disabling  menu item displaying in it. In VS Code [this lines](https://github.com/microsoft/vscode/blob/6fa3243d3facbbf8b87222bcf4e81c5eddf5d47f/src/vs/workbench/contrib/files/browser/fileActions.contribution.ts#L472) in file actions handle that download menu item it seems. We need to disable it or else if possible to edit those code in codeserver instance it might work. But I am not sure.

**Reply by jsjoeio** [reply]
Nice find!! I'll add that to the open issue.

**Comment by jsjoeio** [comment, marked as answer]
This isn't currently supported but something we're planning to work on soon! Follow here: https://github.com/coder/code-server/issues/2426
